testing/src/test/java/io/stargate/it/cql/GeoTypeTest.java

```diff
@@ -0,0 +1,147 @@
+
+    // Creating a new table for each type is too slow, use a single table with all possible types:
+    CreateTable createTableQuery = createTable("test").withPartitionKey("k", DataTypes.INT);
+    for (TypeSample<?> sample : allTypes) {
+      createTableQuery = createTableQuery.withColumn(sample.columnName, sample.cqlType);
+    }
+
+    session.execute(createTableQuery.asCql());
+  }
+
+  @SuppressWarnings("unused") // called by JUnit 5
+  public static List<TypeSample<?>> getAllTypes() {
+    return allTypes;
+  }
+
+  @DisplayName("Should write and read data type")
+  @ParameterizedTest
+  @MethodSource("getAllTypes")
+  public <JavaTypeT> void should_write_and_read(TypeSample<JavaTypeT> sample, CqlSession session) {
+
+    String insertQuery =
+        insertInto("test").value("k", literal(1)).value(sample.columnName, bindMarker()).asCql();
+
+    SimpleStatement simpleStatement = SimpleStatement.newInstance(insertQuery, sample.value);
+    session.execute(simpleStatement);
+    checkValue(sample, session);
+
+    session.execute(BatchStatement.newInstance(BatchType.LOGGED).add(simpleStatement));
+    checkValue(sample, session);
+
+    session.execute(session.prepare(insertQuery).bind(sample.value));
+    checkValue(sample, session);
+  }
+
+  private <JavaTypeT> void checkValue(TypeSample<JavaTypeT> sample, CqlSession session) {
+    String selectQuery =
+        selectFrom("test").column(sample.columnName).whereColumn("k").isEqualTo(literal(1)).asCql();
+    Row row = session.execute(selectQuery).one();
+    assertThat(row).isNotNull();
+    assertThat(row.get(0, sample.javaType)).isEqualTo(sample.value);
+
+    // Clean up for the following tests
+    session.execute("DELETE FROM test WHERE k = 1");
+  }
+
+  private static List<TypeSample<?>> generateAllTypes(CqlIdentifier keyspaceId) {
+    List<TypeSample<?>> primitiveTypes =
+        ImmutableList.of(
+            typeSample(
+                DseDataTypes.POINT, GenericType.of(Point.class), Point.fromCoordinates(1, 2)),
+            typeSample(
+                DseDataTypes.POLYGON,
+                GenericType.of(Polygon.class),
+                Polygon.fromPoints(
+                    Point.fromCoordinates(0, 0),
+                    Point.fromCoordinates(0, 1),
+                    Point.fromCoordinates(1, 1))),
+            typeSample(
+                DseDataTypes.LINE_STRING,
+                GenericType.of(LineString.class),
+                LineString.fromPoints(Point.fromCoordinates(0, 0), Point.fromCoordinates(0, 1))));
+
+    List<TypeSample<?>> allTypes = new ArrayList<>();
+    for (TypeSample<?> type : primitiveTypes) {
+      // Generate additional samples from each primitive
+      allTypes.add(type);
+      allTypes.add(listOf(type));
+      allTypes.add(setOf(type));
+      allTypes.add(mapToIntFrom(type));
+      allTypes.add(mapOfIntTo(type));
+      // TODO: there are some odd codec lookup errors in tupleOfIntAnd(...) and udtOfIntAnd(...)
```

Comment:
To be addressed in a follow-up PR.

Reply:
I think this is because `column.Type.codec()` returns null for the new types.
Unlike collections, tuple and UDT values try to serialize a field as soon as you set it, not when the whole container is serialized.